inspire_utils/name.py

```diff
@@ -173,17 +173,19 @@ def _is_roman_numeral(suffix):
 
         # Create first and middle
         first_name = _ensure_dotted_initials(self.first)
-        middle_name = _ensure_dotted_initials(self.middle)
-
-        if _is_initial(first_name) and _is_initial(middle_name):
-            normalized_names = u'{first_name}{middle_name}'
+        middle_names = [_ensure_dotted_initials(name) for name in self.middle_list]
+
+        if middle_names:
+            if _is_initial(first_name) and _is_initial(middle_names[0]):
+                normalized_names = first_name + middle_names[0]
+            else:
+                normalized_names = u' '.join([first_name, middle_names[0]])
+            for position in range(len(middle_names) - 1):
```

Comment:
if in Python you ever need to manipulate list indices while iterating, you’re doing it wrong. Let's discuss this IRL.